common/src/main/java/io/druid/concurrent/LifecycleLock.java

```diff
@@ -268,12 +275,14 @@ public void exitStop()
   }
 
   /**
-   * Resets the LifecycleLock after {@link #exitStop()}, so that {@link #canStart()} could be called again.
+   * Finalizes stopping the LifecycleLock and resets it, so that {@link #canStart()} could be called again. If this
+   * LifecycleLock is used in a restartable object, this method must be called before exit from stop() on this object,
+   * usually in a finally block.
```

Comment:
The comment after this about IllegalMonitorStateException should refer to canStop() instead of exitStop()

Reply:
Thanks. Updated javadoc.